src/Microsoft.Health.Fhir.SqlServer/Features/Operations/Import/DataGenerator/UriSearchParamsTableBulkCopyDataGenerator.cs

```diff
@@ -0,0 +1,61 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Data;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.SqlServer.Features.Schema.Model;
+using Microsoft.Health.SqlServer.Features.Schema.Model;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Operations.Import.DataGenerator
+{
+    internal class UriSearchParamsTableBulkCopyDataGenerator : SearchParamtersTableBulkCopyDataGenerator
+    {
+        private ITableValuedParameterRowGenerator<IReadOnlyList<ResourceWrapper>, BulkUriSearchParamTableTypeV1Row> _searchParamGenerator;
+
+        internal UriSearchParamsTableBulkCopyDataGenerator()
+        {
+        }
+
+        public UriSearchParamsTableBulkCopyDataGenerator(ITableValuedParameterRowGenerator<IReadOnlyList<ResourceWrapper>, BulkUriSearchParamTableTypeV1Row> searchParamGenerator)
+        {
+            EnsureArg.IsNotNull(searchParamGenerator, nameof(searchParamGenerator));
+
+            _searchParamGenerator = searchParamGenerator;
+        }
+
+        internal override string TableName
+        {
+            get
+            {
+                return VLatest.UriSearchParam.TableName;
+            }
+        }
+
+        internal override void FillDataTable(DataTable table, SqlBulkCopyDataWrapper input)
```

Comment:
>table [](http://example.com/codeflow?start=55&length=5)

Can DataTable be null?

Reply:
This should not be null, add check here for protect. Thanks.